src/contracts/common/ExecutionEnvironment.sol

```diff
@@ -141,6 +141,9 @@ contract ExecutionEnvironment is Base {
             solverTracker.ceiling = solverTracker.etherIsBidToken
                 ? address(this).balance
                 : IERC20(solverOp.bidToken).balanceOf(address(this));
+
+            // Check if the ceiling is sufficient
+            if (solverTracker.ceiling < bidAmount) revert AtlasErrors.InsufficientCeiling();
```

Comment:
Maybe we should call this error BidAmountTooHigh? For context: in invertsBid mode, solvers should compete by giving a _lower_ bid than the other solvers. This lower bid amount represents the amount of tokens they take from the dapp control.

Reply:
Renamed Error